src/Microphysics0M.jl

```diff
@@ -16,28 +16,39 @@ export remove_precipitation
 
 """
     remove_precipitation(params_0M, q; q_vap_sat)
+    remove_precipitation(params_0M, q_liq, q_ice; q_vap_sat)
 
  - `params_0M` - a struct with 0-moment parameters
- - `q` - current PhasePartition
+ - `q` - current Thermodynamics.PhasePartition or `q_liq` and `q_ice` specific contents
  - `q_vap_sat` - specific humidity at saturation
 
 Returns the `q_tot` tendency due to the removal of precipitation.
 The tendency is obtained assuming a relaxation with a constant timescale
 to a state with precipitable water removed.
 The threshold for when to remove `q_tot` is defined either by the
 condensate specific content or supersaturation.
-The thresholds and the relaxation timescale are defined in
-ClimaParams.
+The thresholds and the relaxation timescale are defined `Parameters0M` struct.
 """
 remove_precipitation(
     (; τ_precip, qc_0)::CMP.Parameters0M,
     q::TD.PhasePartition,
 ) = -max(0, (q.liq + q.ice - qc_0)) / τ_precip
```

Comment:
```suggestion
remove_precipitation(
    p::CMP.Parameters0M,
    q::TD.PhasePartition,
) = remove_precipitation(p, q.liq, q.ice)
```

could also use this to reduce code paths

Reply:
+ reduce need for additional testing

Reply:
Sure, that also works. Tbh, I think I can/will get rid of `PhasePartition` on the `Atmos` side and just support the simple version of passing in individual tracers.